controllers/kuadrant_controller.go

```diff
@@ -386,14 +394,27 @@ func (r *KuadrantReconciler) createOnlyInKuadrantNSCb(ctx context.Context, kObj
 			return err
 		}
 
-		k8sObjKind := k8sObj.DeepCopyObject().GetObjectKind()
+		var newObj client.Object
+		newObj = k8sObj
+
+		switch obj := k8sObj.(type) {
+		case *appsv1.Deployment: // If it's a Deployment obj, it adds the required env vars
+			obj.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{
+				{Name: envLimitadorNamespace, Value: kObj.Namespace},
```

Comment:
Should we consider to pass only Limitador CR name/NS?? According to [this](https://github.com/Kuadrant/limitador-operator/issues/21) and if we implement that, the kuadrant controller (the control plane not the operator) only needs to know the location of the Limitador CR. It will be the Limitador Operator the one responsible of injecting  limits into pods.